XenAdmin/SettingsPanels/PerfmonAlertOptionsPage.cs

```diff
@@ -150,8 +141,8 @@ public void SetXenObjects(IXenObject orig, IXenObject clone)
             _OrigSmtpServerAddrTextBox = SmtpServerAddrTextBox.Text;
             _OrigSmtpServerPortTextBox = SmtpServerPortTextBox.Text;
 
-            if (_bSupportMailLanguage)
-                _OrigMailLanguageCodeComboBox = null == MailLanguageComboBox.SelectedValue ? null : MailLanguageComboBox.SelectedValue.ToString();
+            if (_bSupportMailLanguage && MailLanguageComboBox.SelectedValue != null)
+                _OrigMailLanguageCodeComboBox = MailLanguageComboBox.SelectedValue.ToString();

```

Comment:
I propose to rename `_OrigMailLanguageCodeComboBox` to `_OrigMailLanguageCode` and save the original code into it (not the combobox's selected value).
So this code becomes:

```
if (_bSupportMailLanguage && _PerfmonOptions != null)
    _OrigMailLanguageCode = _PerfmonOptions.MailLanguageCode;
else
    _OrigMailLanguageCode = null;
```
Then, on the `HasChanged` function below, we will have:
 `(_bSupportMailLanguage && _OrigMailLanguageCode != MailLanguageComboBox.SelectedValue.ToString())`